### PR TITLE
change state with redux

### PR DESCRIPTION
**Pull request by Eyyan-Amir**
using redux

### CODE REVIEW

File: src/action/action.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import axios from "axios";
```
Comment (saqlainrasheed) [inline src/action/action.tsx]: Create separate folder for the redux and put all actions there

File: src/action/action.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+export const setMovies = (movieList: any) => {
+	return {
+		type: "SET_MOVIE",
```
Comment (saqlainrasheed) [inline src/action/action.tsx]: put type in constants 
i.e

export const SET_MOVIES = 'SET_MOVIES'

File: src/action/action.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+export const setGenre = (movieList: any) => {
+	return {
+		type: "SET_GENRE",
+		payload: movieList,
+	};
+};
+
+export const setSearchMovies = (movieList: any) => {
```
Comment (saqlainrasheed) [inline src/action/action.tsx]: setSearchResults

File: src/action/action.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import axios from "axios";
+
+export const setMovies = (movieList: any) => {
```
Comment (saqlainrasheed) [inline src/action/action.tsx]: moviesList

File: src/action/action.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+export const setSearchMovies = (movieList: any) => {
+	return {
+		type: "SET_SEARCH_MOVIE",
+		payload: movieList,
+	};
+};
+
+export const setMovieDetail = (movieList: any) => {
```
Comment (saqlainrasheed) [inline src/action/action.tsx]: (movieDetails:any) =>

File: src/action/action.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+	return {
+		type: "SET_DETAIL_MOVIE",
+		payload: movieList,
+	};
+};
+
+export const addComments = (items: any) => {
```
Comment (saqlainrasheed) [inline src/action/action.tsx]: addeComment = (comment: string)

File: src/action/action.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+	return {
+		type: "ADD_COMMENTS",
+		payload: items,
+	};
+};
+
+export const setRegistration = (items: any) => {
```
Comment (saqlainrasheed) [inline src/action/action.tsx]: setUser = (user: <make interface>)

File: src/components/Registration.tsx [hunk 8]
```diff
@@ -1,8 +1,10 @@
 import { Form, Formik, Field, ErrorMessage } from "formik";
-import { useState, useEffect } from "react";
+import { useEffect } from "react";
 import { Link } from "react-router-dom";
 import * as yup from "yup";
 import ErrorText from "./common/ErrorText";
+import { useSelector, useDispatch } from "react-redux";
```
Comment (saqlainrasheed) [inline src/components/Registration.tsx:6]: Use dispatch and connect methods instead

File: src/components/common/MovieDetail.tsx [hunk 9]
```diff
@@ -1,29 +1,29 @@
-import { useState, useEffect } from "react";
+import { useEffect } from "react";
 import { useParams } from "react-router-dom";
 import * as yup from "yup";
 import axios from "axios";
+import { Link } from "react-router-dom";
 import CommentsForm from "./CommentsForm";
+import { useSelector, useDispatch } from "react-redux";
```
Comment (saqlainrasheed) [inline src/components/common/MovieDetail.tsx]: use connect method in 
search, home page and comments

File: src/components/common/MovieDetail.tsx [hunk 10]
```diff
@@ -1,29 +1,29 @@
...
-		homepage: "",
-	};
-
 	const initialValue: CommentType = { comment: "" };
 
 	const { id } = useParams();
 
-	const [items, setItems] = useState<CommentType[]>([]);
-	const [detailMovie, setDetailMovie] = useState(movieDetail);
+	const { detailMovie, items } = useSelector(
```
Comment (saqlainrasheed) [inline src/components/common/MovieDetail.tsx]: useSelector<interface>

Comment (saqlainrasheed) [inline src/components/common/MovieDetail.tsx]: movieDetails

File: src/components/common/SearchResult.tsx [hunk 11]
```diff
@@ -15,24 +17,28 @@ interface SearchResultType {
 
 export default function SearchResult() {
 	const { name } = useParams();
-	const [serachMovie, setSearchMovie] = useState<string | any>(name);
-	const [movies, setMovies] = useState<SearchResultType[]>([]);
+	const { searchMovies } = useSelector(
```
Comment (saqlainrasheed) [inline src/components/common/SearchResult.tsx]: searchResults

File: src/reducer/reducer.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+			return {
+				...state,
+				genreMovies: payload,
+			};
+		case "SET_SEARCH_MOVIE":
+			return {
+				...state,
+				searchMovies: payload,
+			};
+		case "SET_DETAIL_MOVIE":
```
Comment (saqlainrasheed) [inline src/reducer/reducer.tsx]: SET_MOVIE_DETAILS

**Comment by Eyyan-Amir** [comment]
change state with redux

**Pull request merged**